src/main/java/net/atos/zac/flowable/TaakVariabelenService.java

```diff
@@ -130,15 +133,29 @@ public void setTaakdocumenten(final Task task, final List<UUID> taakdocumenten)
     }
 
     public UUID readZaakUUID(final TaskInfo taskInfo) {
-        return (UUID) readVariable(taskInfo, VAR_ZAAK_UUID);
+        Object obj = readVariable(taskInfo, VAR_ZAAK_UUID);
+        return switch (obj) {
+            case UUID uuid -> uuid;
+            case Object o -> {
```

Comment:
Before this code I would expect that this method resulted in a ClassCastException because it could cast the String to a UUID in these cases? But these ClassCastExceptions were not logged or did not give enough information? Normally I would say: a ClassCastException is fine in these cases and here in this code you could catch them and embed them in a `IllegalArgumentException`?

Reply:
We can do that indeed. My initial idea was to handle String too (if needed) and convert it to UUID.

The only thing missing from ClassCastException is the actual object value. Not sure how much value we get out of this now.

We can simply revert back to ClassCast and just change the test expectations

Reply:
Reverted to using ClassCast. Code is much simpler this way